pkg/traceql/combine.go

```diff
@@ -326,6 +326,10 @@ type QueryRangeCombiner struct {
 	req     *tempopb.QueryRangeRequest
 	eval    *MetricsFrontendEvaluator
 	metrics *tempopb.SearchMetrics
+
+	maxSeries        int
```

Comment:
you only need 2 of these 3, right?

why not remove maxSeriesReached and  change the function below to
```
func (q *QueryRangeCombiner) MaxSeriesReached() bool {
	return q.seriesCount >= q.maxSeries
}
```